crates/wash-cli/src/up/mod.rs

```diff
@@ -619,6 +622,102 @@ pub async fn handle_up(cmd: UpCommand, output_kind: OutputKind) -> Result<Comman
     Ok(CommandOutput::new(out_text, out_json))
 }
 
+/// Check if a wasmcloud host is running
+async fn running_host_count(wasmcloud_opts: WasmcloudOpts, install_dir: PathBuf) -> Result<usize> {
+    let ctl_client = wasmcloud_opts.clone().into_ctl_client(None).await?;
+    let host_count = ctl_client
+        .get_hosts()
+        .await
+        .map_err(|e| anyhow!(e))?
+        .into_iter()
+        .filter_map(|r| r.response)
+        .count();
+    if host_count > 0 {
+        return Ok(host_count);
+    }
+    // Wasmcloud host might be starting but not up yet. Check if the process in the pid file is running.
+    let pid_file_string = tokio::fs::read_to_string(&install_dir.join(WASMCLOUD_PID_FILE)).await?;
+    let pid_file_value: Value = serde_json::from_str(&pid_file_string)?;
+    if let Some(pid) = pid_file_value.get("pid") {
+        if is_process_running(&pid.to_string()) {
+            return Ok(1);
+        }
+    }
+    Ok(0)
+}
+
+/// Check is process is running
+fn is_process_running(pid: &str) -> bool {
+    match pid.parse() {
+        Ok(pid) => {
+            let mut sys = System::new_all();
+            sys.refresh_processes();
+            sys.processes().get(&pid).is_some()
+        }
+        Err(_) => false,
+    }
+}
+
+/// Spawn off a task that waits until the host has started,
+/// then loads and deploys the WADM manifest.
+async fn process_wasm_manifest(
```

Comment:
Should this be `process_wadm_manifest`?

Reply:
Heads up @cdmurph32 this is still named with `wasm` instead of `wadm`

Reply:
Fixed. Thanks.